lib/wasabi/parser.rb

```diff
@@ -319,14 +319,13 @@ def section(section_name)
     end
 
     def sections
-      return @sections if @sections
-
-      sections = {}
-      document.root.element_children.each do |node|
-        (sections[node.name] ||= []) << node
-      end
-
-      @sections = sections
+      @sections ||= begin
```

Comment:
@chaymaeBZ @olleolleolle To simplify this code, we could use the [Enumerable#group_by](https://apidock.com/ruby/Enumerable/group_by) method... Wdyta?

```rb
@sections ||= document.root.element_children.group_by { |node| node.name }
```

Reply:
I think that sounds fine, if you make a PR, you can see if all tests still pass, would be nice.

Reply:
@olleolleolle Done: https://github.com/savonrb/wasabi/pull/90